cpu/cpu.go

```diff
@@ -182,26 +196,43 @@ func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.Metri
 		p.snapMetricsNames, p.procStatMetricsNames); err != nil {
 		return nil, err
 	}
+	ts := time.Now()
+	prefix := filepath.Join(vendor, fs, pluginName)
 	for _, metricType := range metricTypes {
 		ns := metricType.Namespace()
 		if len(ns) != maxNamespaceSize {
 			return nil, fmt.Errorf("Incorrect namespace length (len = %d)", len(ns))
 		}
-
-		val, err := getMapValueByNamespace(p.stats[ns.Strings()[3]], ns.Strings()[4:])
-
-		if err != nil {
-			return metrics, err
-		}
-
-		metric := plugin.MetricType{
-			Namespace_: ns,
-			Data_:      val,
-			Timestamp_: time.Now(),
+		if ns[len(ns)-2].Value == "*" {
+			for cpuId, cpuStats := range p.stats {
+				for k, v := range cpuStats {
+					if ns[len(ns)-1].Value == k && v != nil {
+						ns1 := core.NewNamespace(strings.Split(prefix, string(os.PathSeparator))...).
+							AddStaticElement(cpuId).
```

Comment:
@IzabellaRaulin Thanks for pointing this out. I am not 100% sure of what this might really mean in depth but will have a look at the code you mention to clarify my mind on the matter.

Please note that as mentioned in [this other PR](https://github.com/intelsdi-x/snap-plugin-collector-swap/pull/10#discussion_r75105370) I have submitted other PRs with most probably the same kind of code :-( so if you are right, I might need to rework a little bit on these too, but this is no pb at all

@jcooklin your inputs on all this would be great help, thanks in advance